### PR TITLE
Bugfix FXIOS-4988 [v124] "Request Desktop Site" not sufficiently presented for accessibility

### PR DESCRIPTION
**Pull request by imsanchit**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-2110)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/4988)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: firefox-ios/Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -4703,6 +4703,11 @@ extension String {
         tableName: nil,
         value: "Add to Reading List",
         comment: "Accessibility label for action adding current page to reading list.")
+    public static let TabLocationReloadButtonMoreOptionsAccessibilityLabel = MZLocalizedString(
+        key: "Address.Bar.MoreOptions",
+        tableName: nil,
+        value: "More Options",
+        comment: "Accessibility label for action showing more options")
```
Comment (nbhasin2) [inline firefox-ios/Client/Frontend/Strings.swift:4710]: @thatswinnie @adudenamedruby do you know if for such items you ever talked with accessibility team or anyone for adding values like these?

Comment (adudenamedruby) [reply firefox-ios/Client/Frontend/Strings.swift:4710]: As far as I know, copy should be providing strings for strings for accessibility.

Also, @imsanchit, please take a look at the wiki for adding strings. New strings will require a specific format. :)

Comment (imsanchit) [reply firefox-ios/Client/Frontend/Strings.swift]: @nbhasin2 @adudenamedruby Please note these strings aren't suggested by anyone, I have added them as per my understanding as a part of this fix, we can change these values as per your suggestions. Also, I have updated the PR with the required format, please have a look and feel free to point out any required corrections.

**Comment by adudenamedruby** [comment]
summoning @ewachowiak to check copy for strings for accessibility here.

**Comment by ewachowiak** [comment]
Can you help me understand the actions for which we need strings? For example, providing a list of the user actions and VoiceOver contexts that would require new strings. Thanks.

**Comment by adudenamedruby** [comment]
@ewachowiak looks like @imsanchit is adding the custom action to the reload button, as the ticket is suggesting.

**Comment by ewachowiak** [comment]
Can you please be more specific about what you mean by "custom action"? I read the ticket and it wasn't clear to me. Please walk me through what the user is doing (sounds like selecting the refresh icon) and what Voiceover is telling them when they arrive at this moment (is it already saying something like "menu" for example?). Also, is it just one string you need, or multiple? For which actions specifically? 

Please feel free to set up a brief 1:1 conversation if it is easier to explain in person.

**Comment by ewachowiak** [comment]
I checked in with my manger: the content design team is not staffed to be writing voiceover text, just hover text and tooltips. My advice for this and future cases is to have the eng team work with the a11y team to write these. I am happy to review already-written voiceover strings if there is a question or you want another set of eyes. 

For this specific request, @imsanchit or @adudenamedruby can you please write the strings and let me know if you need a reviewer?

File: firefox-ios/Client/Frontend/Strings.swift [hunk 2]
```diff
@@ -4826,6 +4831,11 @@ extension String {
         tableName: nil,
         value: nil,
         comment: "Accessibility label for the navigation toolbar displayed at the bottom of the screen.")
+    public static let TabToolbarMoreOptionsAccessibilityLabel = MZLocalizedString(
+        key: "Toolbar.Accessibility.MoreOptions.v123",
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Strings.swift:4835]:
```suggestion
        key: "Toolbar.Accessibility.MoreOptions.v124",
```

**Review by adudenamedruby — COMMENTED** [review]
123 string freeze for translations has passed a few weeks ago, so these'll be for 124. :)

**Comment by adudenamedruby** [comment]
@ewachowiak both strings read "More options"

**Comment by thatswinnie** [comment]
> @thatswinnie, just double checking with you that this seems fine. :)

While this certainly works, I think it would be better to add `Request Desktop Site` as a custom action directly instead of making the user jump through the hoop of selecting the custom action `More Options` and then `Request Desktop Site`. @anna-yeddi would you agree?

**Comment by anna-yeddi** [comment]
> > @thatswinnie, just double checking with you that this seems fine. :)
> 
> While this certainly works, I think it would be better to add `Request Desktop Site` as a custom action directly instead of making the user jump through the hoop of selecting the custom action `More Options` and then `Request Desktop Site`. @anna-yeddi would you agree?

Thank you for tagging and pinging me, @thatswinnie! Apologies for the delay.

This action is already available with VoiceOver - you could focus the `Reload` button and then double-tap and hold (this is standard VO gesture that is equal to default tap and hold aka long-press). I checked with the home screen apps, the VoiceOver does not provide any additional hint (i.e. TalkBack on Android announces `actions available` - but this is provided by the screen reader itself), thus technically we do not have to provide anything extra. The gesture works - this is important.

That's being said, I agree with the reporter that it is not easily discoverable. I did not know about this shortcut until today too :) We could provide `accessibilityHint` like `More options`. This to be applied to the `Reload` button itself, not to the entire toolbar since the `Reload` should be focused and double-tapped and held to open the menu, not the toolbar. Bonus, we could reuse this string for hints to other places where we hide this context menu too.

I hope it helps, LMK if more info or other ideas could be useful (I've fixed GH tracking, I think)

**Comment by anna-yeddi** [comment]
And I think, we technically can do a custom action for the long-tap and, with VO, double-tap and hold gesture to just go directly to the `Request desktop site`, but we'd need to make the same gesture for both users of touch and users of VoiceOver (aka not showing the menu element, but directly activating it). I am not sure this would be very intuitive though. But if we choose to go this route, we could use the same `accessibilityHint` to `double-tab and hold to Request Desktop Site` so it's clear what to do and what'll happen

**Comment by adudenamedruby** [comment]
@thatswinnie, thoughts on Anna's solution?

**Comment by thatswinnie** [comment]
@adudenamedruby we should not make things more complicated by adding a custom action. So as this action can already be triggered in voice over using double-tab and hold we should only add an accessibility hint to the reload button. @ewachowiak could you provide us with a string for the accessibility hint for this? Would `More options available` or `double-tap and hold for More Options` be ok?

**Comment by ewachowiak** [comment]
@thatswinnie "Double tap and hold for more options" seems the most clear, provided that that sequence of actions is accurate. CC @anna-yeddi for verification.

**Comment by anna-yeddi** [comment]
> @thatswinnie "Double tap and hold for more options" seems the most clear, provided that that sequence of actions is accurate. CC @anna-yeddi for verification.

@thatswinnie thumbs up on the `Double tap and hold for more options` option :) Thank you all for the investigation!

File: firefox-ios/Client/Frontend/Strings.swift [hunk 3]
```diff
@@ -4850,6 +4850,11 @@ extension String {
         tableName: nil,
         value: nil,
         comment: "Accessibility label for the reload button")
+    public static let TabLocationReloadAccessibilityHint = MZLocalizedString(
+        key: "Address.Bar.Reload.A11y.Hint.v124",
+        tableName: "Reload Button",
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Strings.swift:4855]:
```suggestion
        tableName: "TabLocation",
```

**Review by adudenamedruby — APPROVED** [review]

**Pull request merged**